docs/source/contribution/cpu_features.rst

```diff
@@ -0,0 +1,218 @@
+
+Suppose that you are working on some algorithm ``Abc`` in |short_name|.
+
+The part of the implementation of this algorithms that is running on CPU should be located in
+`cpp/daal/src/algorithms/abc` folder.
+
+Suppose that it provides:
+
+- ``classification`` and ``regression`` learning tasks;
+- ``training`` and ``inference`` stages;
+- ``method1`` and ``method2`` for the ``training`` stage and only ``method1`` for ``inference`` stage;
+- only ``batch`` computational mode.
+
+Then the `cpp/daal/src/algorithms/abc` folder should contain at least the following files:
+
+::
+
+  cpp/daal/src/algorithms/abc/
+    |-- abc_classification_predict_method1_batch_fpt_cpu.cpp
+    |-- abc_classification_predict_method1_impl.i
+    |-- abc_classification_predict_kernel.h
+    |-- abc_classification_train_method1_batch_fpt_cpu.cpp
+    |-- abc_classification_train_method2_batch_fpt_cpu.cpp
+    |-- abc_classification_train_method1_impl.i
+    |-- abc_classification_train_method2_impl.i
+    |-- abc_classification_train_kernel.h
+    |-- abc_regression_predict_method1_batch_fpt_cpu.cpp
+    |-- abc_regression_predict_method1_batch_fpt_cpu.cpp
+    |-- abc_regression_predict_method1_impl.i
+    |-- abc_regression_predict_kernel.h
+    |-- abc_regression_train_method1_batch_fpt_cpu.cpp
+    |-- abc_regression_train_method2_batch_fpt_cpu.cpp
+    |-- abc_regression_train_method1_impl.i
+    |-- abc_regression_train_method2_impl.i
+    |-- abc_regression_train_kernel.h
+
+Alternative variant of the folder structure to avoid storing too many files within a single folder
+could be:
+
+::
+
+  cpp/daal/src/algorithms/abc/
+    |-- classification/
+    |     |-- abc_classification_predict_method1_batch_fpt_cpu.cpp
+    |     |-- abc_classification_predict_method1_impl.i
+    |     |-- abc_classification_predict_kernel.h
+    |     |-- abc_classification_train_method1_batch_fpt_cpu.cpp
+    |     |-- abc_classification_train_method2_batch_fpt_cpu.cpp
+    |     |-- abc_classification_train_method1_impl.i
+    |     |-- abc_classification_train_method2_impl.i
+    |     |-- abc_classification_train_kernel.h
+    |-- regression/
+          |-- abc_regression_predict_method1_batch_fpt_cpu.cpp
+          |-- abc_regression_predict_method1_impl.i
+          |-- abc_regression_predict_kernel.h
+          |-- abc_regression_train_method1_batch_fpt_cpu.cpp
+          |-- abc_regression_train_method2_batch_fpt_cpu.cpp
+          |-- abc_regression_train_method1_impl.i
+          |-- abc_regression_train_method2_impl.i
+          |-- abc_regression_train_kernel.h
+
+The names of the files stay the same in this case, just the folder layout differs.
+
+Further the purpose and contents of each file are to be described on the example of classification
+training task. For other types of the tasks the structure of the code is similar.
+
+\*_kernel.h
+-----------
+
+Those files contain the definitions of one or several template classes that define member functions that
```

Comment:
nit: Don't start the section with a pronoun. Put a full description of what you are describing. Maybe:

```
In the directory structure introduced in the last section, there are files with a `_kernel.h` suffix. These contain the definitions of ...
```